src/utils/service/service.js

```diff
@@ -149,31 +150,75 @@ class Service {
    *
```

Comment:
The four CRUD methods here are virtually identical - could they be DRYed a little? There might be issues with `this` if you're using a generic function, but there should be ways to deal with this.
Have considered a single method that is called by the 4 CRUD methods?

Reply:
Maybe we could, but for the additions I have made I expect the `else`/`if` situation should not exist too long, then maybe we could refactor if see the need.